packages/ragbits-document-search/src/ragbits/document_search/_main.py

```diff
@@ -150,16 +150,29 @@ async def insert_elements(self, elements: list[Element]) -> None:
         Args:
             elements: The list of Elements to insert.
         """
-        vectors = await self.embedder.embed_text([element.key for element in elements])
+        elements_with_text = [element for element in elements if element.key]
+        images_with_text = [element for element in elements_with_text if isinstance(element, ImageElement)]
+        vectors = await self.embedder.embed_text([element.key for element in elements_with_text if element.key])
 
         image_elements = [element for element in elements if isinstance(element, ImageElement)]
-        entries = [element.to_vector_db_entry(vector) for element, vector in zip(elements, vectors, strict=False)]
+
+        num_images_with_no_textual_repr = len(image_elements) - len(images_with_text)
+        if num_images_with_no_textual_repr > 0:
+            warnings.warn(
```

Comment:
According to the ticket, this was supposed to be individual warnings for images that don't have ANY kind of embedding created.

As is, this will warn about ALL images in the scenario in which the user only want to create visual embeddings - which is a normal scenario, which we want to fully support.

Reply:
done

Reply:
but isn't it a case that for example we define some processor that we intent to describe and / or ocr the image, but for some case it fails - api error, etc. shouldn't we somehow inform user about this case ?

Reply:
If there is an API error in processor than the processor should indicate that for the user. On the other hand, having a processor that doesn't add text descriptions for images is described in the ticket (and will be described in the documentation) as the main way of configuring the system to only create visual embedding for images.

Reply:
ok will remove that however we also have a warning here for not supported vision embedding if the embedder does not support vision - would also remove it and warn only about not embedded at all images individually

Reply:
done